File: pset6/readability/readability.py

text = input("Text: ")
word_count = 1
letter_count = 0
sentence_count = 0

length=len(text)

for i in range(length):
    if text[i] == " ":
        word_count += 1
    elif text[i] in [".", "?", "!"]:
        sentence_count += 1
    elif ord(text[i]) in range(65, 91) or ord(text[i]) in range(97, 123):
        letter_count += 1

index_w = 100 / word_count



avg_l = letter_count * index_w
avg_s = sentence_count * index_w


coleman_liau = 0.0588 * avg_l - 0.296 * avg_s - 15.8
result = round(coleman_liau)

if result < 1:
    print("Before Grade 1")
elif result >= 16:
    print(f"Grade 16+")
else:
    print(f"Grade {result}")

print(f"{sentence_count} sentences. {word_count} words and {letter_count} letters.")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    //prompting user for text
    string txt = get_string("Text:");
    //initiating counters. Word is assigned to 1, since the last one wouldn't be counted as it has no space after
    int lettercount = 0;
    float wordcount = 1;
    int sentences = 0;
    //treating the string as an array of chars, using a loop + bool to check if it's letter, space or punctuation
    for (int i = 0; i < strlen(txt); i++)
    {
        if ((txt[i] >= 65 && txt[i] <= 90) || (txt[i] >= 97 && txt[i] <= 122))
        {
            lettercount++;
        }

        else if (txt[i] == 32)
        {
            wordcount++;
        }

        else if (txt[i] == 46 || txt[i] == 33 || txt[i] == 63)
        {
            sentences++;
        }
    }
//getting the value that makes wordcount == 100. Multiplying letters and sentences by Y to get the avg per 100
    float Y = 100 / wordcount;
    float L = Y * lettercount;
    float S = Y * sentences;
//Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);
//printing out the correct value
    if (index > 0 && index < 16)
    {
        printf("Grade %.0f\n", index);
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index <= 0)
    {
        printf("Before Grade 1\n");

    }
}
